src/context/DisableLunaUstWarning/index.tsx

```diff
@@ -0,0 +1,67 @@
+import noop from 'noop-ts';
+import React, { useContext, useEffect, useMemo, useState } from 'react';
+
+import { useGetUserMarketInfo } from 'clients/api';
+import { LunaUstWarningModal } from 'components/LunaUstWarningModal';
+import { TOKENS } from 'constants/tokens';
+import { AuthContext } from 'context/AuthContext';
+
+export interface DisableLunaUstWarningContextValue {
+  hasLunaOrUstCollateralEnabled: boolean;
+  shouldShowLunaUstWarningModal: boolean;
+  openLunaUstWarningModal: () => void;
+  closeLunaUstWarningModal: () => void;
+}
+
+export const DisableLunaUstWarningContext = React.createContext<DisableLunaUstWarningContextValue>({
+  hasLunaOrUstCollateralEnabled: false,
+  shouldShowLunaUstWarningModal: false,
+  openLunaUstWarningModal: noop,
+  closeLunaUstWarningModal: noop,
+});
+
+export const DisableLunaUstWarningProvider: React.FC = ({ children }) => {
+  const { account } = useContext(AuthContext);
+  const accountAddress = account?.address || '';
+  const {
+    data: { assets },
+  } = useGetUserMarketInfo({
+    accountAddress,
+  });
+
+  const [shouldShowLunaUstWarningModal, setShouldShowModal] = useState(false);
```

Comment:
Should implies a question or an element of uncertainty. 
```suggestion
  const [showLunaUstWarningModal, setShowModal] = useState(false);
```

Reply:
I disagree. `showLunaUstWarningModal` sounds like a function name to me, I'd expect it to be called like this:
```
showLunaUstWarningModal();
```

We can change `shouldShowLunaUstWarningModal` if you think it's unclear what it represents, but my goal with this variable is to emulate the same logic as with `isLoading` for example: it implies a state and a boolean.

Reply:
should is not clear. You could also think it is callable and that you would receive a variable back telling you whether the modal should be shown or not. I only removed the word `should` as the main comment is that it is not a precise word for describing state but conveys additional connotations that aren't applicable here

Reply:
I see. How about `isLunaUstWarningModalOpen` then?